packages/dendron-plugin-views/src/components/DendronApp.tsx

```diff
@@ -161,14 +166,45 @@ function DendronVSCodeApp({ Component }: { Component: DendronComponent }) {
         }
         break;
       }
-
+      case NoteViewMessageEnum.onLock: {
+        logger.info({ ctx, msg: "onLock" });
+        ideDispatch(ideSlice.actions.setLock(true));
+        break;
+      }
+      case NoteViewMessageEnum.onUnlock: {
+        logger.info({ ctx, msg: "onUnlock" });
+        ideDispatch(ideSlice.actions.setLock(false));
+        break;
+      }
       default:
         logger.error({ ctx, msg: "unknown message", payload: msg });
         break;
     }
   });
 
-  return <Component {...props} />;
+  const isLocked = props.ide.isLocked;
+
+  const handleLock = () => {
+    postVSCodeMessage({
+      type: NoteViewMessageEnum.onUnlock,
+      data: {},
+      source: DMessageSource.webClient,
+    });
+  };
+
+  return (
+    <>
+      <Component {...props} />
+      {isLocked && (
```

Comment:
to make this discoverable, can we have the lock be transparent if not locked so that users can toggle by clicking on it? its okay to split this out as a separate pr

Reply:
will catch up on that here https://github.com/dendronhq/dendron/tree/enhance/toggle-preview-lock-from-inside-preview